pex/scie/science.py

```diff
@@ -0,0 +1,320 @@
+        installed_pex_dir = os.path.join(pex_root, unzip_dir_relpath(pex_hash))
+
+    env_replace = {
+        "PEX_ROOT": pex_root,
+    }
+    env = {
+        "remove_re": {"PEX_.*"},
+        "replace": env_replace,
+    }
+
+    lift = {
+        "name": name,
+        "ptex": {
+            "id": "ptex",
+            "version": PTEX_VERSION,
+            "argv1": "{scie.env.PEX_BOOTSTRAP_URLS={scie.lift}}",
+        },
+        "scie_jump": {"version": SCIE_JUMP_VERSION},
+        "files": [{"name": "configure-binding.py"}, {"name": "pex"}],
+        "commands": [
+            {
+                "env": env,
+                "exe": "{scie.bindings.configure:PYTHON}",
+                "args": ["{scie.bindings.configure:PEX}"],
+            }
+        ],
+        "bindings": [
+            {
+                "env": dict(
+                    env,
+                    replace=dict(
+                        env_replace,
+                        PEX_INTERPRETER="1",
+                        _PEX_SCIE_INSTALLED_PEX_DIR=installed_pex_dir,
+                        # We can get a warning about too-long script shebangs, but this is not
+                        # relevant since we above run the PEX via python and not via shebang.
+                        PEX_EMIT_WARNINGS="0",
+                    ),
+                ),
+                "name": "configure",
+                "exe": "#{cpython:python}",
+                "args": ["{pex}", "{configure-binding.py}"],
+            }
+        ],
+    }  # type: Dict[str, Any]
+
+    for target in configuration.targets:
+        manifest_path = os.path.join(
+            safe_mkdtemp(),
+            target.platform.qualified_file_name("{name}-lift.toml".format(name=name)),
+        )
+        with safe_open(manifest_path, "w") as fp:
+            toml.dump(
+                {
+                    "lift": dict(
+                        lift,
+                        platforms=[target.platform.value],
+                        interpreters=[
+                            {
+                                "id": "cpython",
+                                "provider": "PythonBuildStandalone",
+                                "release": target.pbs_release,
+                                "version": target.version_str,
+                                "lazy": configuration.options.style is ScieStyle.LAZY,
+                            }
+                        ],
+                    )
+                },
+                fp,
+            )
+        yield Manifest(target=target, path=manifest_path)
+
+
+def _science_dir(
+    env,  # type: Variables
+    *components  # type: str
+):
+    # type: (...) -> str
+    return os.path.join(env.PEX_ROOT, "scies", "science", str(MIN_SCIENCE_VERSION), *components)
+
+
+def _qualified_science_binary_name():
+    # type: () -> str
+    return SciePlatform.current().qualified_binary_name("science")
+
+
+def _science_binary_names():
```

Comment:
Last note before I crash, should the fat `science` be downloaded instead of the ptex'd one? It's being downloaded with the intent of immediate usage.

What I don't know offhandedly is whether the lazy `science` binary can be leveraged to save a few megs off future updates.

Reply:
I think it does make sense to use the fat binary - but more because it provides exactly 1 download URL and since Pex tries to support firewall use cases, I only need to add 1 option `--scie-science-binary <URL>` to complement the existing `PEX_BOOTSTRAP_URLS` escape hatch for the PBS downloads. I'll fix that and add the option.

Reply:
Fixed to fat and now sporting `--scie-science-binary <FILE or URL>`.